phases/05-nlp-foundations-to-advanced/15-topic-modeling/code/main.py

```
@@ -0,0 +1,99 @@
+import random
+import re
+
+
+def tokenize(text):
+    return [t for t in re.findall(r"[a-z]+", text.lower()) if len(t) > 2]
+
+
+def collapsed_gibbs_lda(docs, n_topics, n_iters=200, alpha=0.1, beta=0.01, seed=0):
+    rng = random.Random(seed)
```

Ruff (0.15.10)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)